src/get_trigger.ts

```diff
@@ -0,0 +1,52 @@
+import { parse, path } from "./deps.ts";
+
+const getTrigger = async () => {
+  const source = parse(Deno.args).source as string;
+
+  if (!source) throw new Error("A source path needs to be defined");
+
+  const fullPath = path.isAbsolute(source)
+    ? source
+    : path.join(Deno.cwd(), source || "");
+
+  return await readFile(fullPath);
+};
+
+const readFile = async (path: string) => {
+  try {
+    const { isFile } = await Deno.stat(path);
+    if (!isFile) throw new Error("The specified source is not a valid file.");
+    if (path.endsWith(".json")) return readJSONFile(path);
+    return readTSFile(path);
+  } catch (e) {
+    if (e instanceof Deno.errors.NotFound) {
+      throw new Error("Trigger Definition file cannot be found");
+    }
+    throw e;
+  }
+};
+
+const readJSONFile = async (path: string) => {
```

Comment:
What do you think about re-using the `getJSON` helper under `src/utilities.ts`? Totally optional, esp. since we discussed with @misscoded about changing this utility method's behaviour (as described in the TODO near that code).

Reply:
I'm happy to adopt that when it's formalized. Like the TODO says we need it to bubble up the error for it to hit the `catch` above so we can log something that makes sense to the hook instance